src/main/java/com/online/cinema/dao/GenericDaoImpl.java

```diff
@@ -1,49 +1,43 @@
-package com.online.cinema.dao.impl;
+package com.online.cinema.dao;
 
-import com.online.cinema.dao.MovieDao;
 import com.online.cinema.exceptions.DataProcessingException;
-import com.online.cinema.lib.Dao;
-import com.online.cinema.model.Movie;
 import com.online.cinema.util.HibernateUtil;
 import java.util.List;
 import javax.persistence.criteria.CriteriaQuery;
 import org.hibernate.Session;
 import org.hibernate.Transaction;
 
-@Dao
-public class MovieDaoImpl implements MovieDao {
-
+public abstract class GenericDaoImpl<T> implements GenericDao<T> {
     @Override
-    public Movie add(Movie movie) {
+    public T add(T element) {
         Transaction transaction = null;
         Session session = null;
         try {
             session = HibernateUtil.getSessionFactory().openSession();
             transaction = session.beginTransaction();
-            session.save(movie);
+            session.save(element);
             transaction.commit();
-            return movie;
+            return element;
         } catch (Exception e) {
             if (transaction != null) {
                 transaction.rollback();
             }
-            throw new DataProcessingException("Can't insert Movie entity", e);
+            throw new DataProcessingException("Can't insert entity", e);
```

Comment:
when you re-throw exception it's better that you give some info about entity, can we try to use element.toString in message?